bin/zerobufCxx.py

```diff
@@ -362,14 +362,17 @@ def get_declaration(self):
 
     def from_json(self):
         if self.value_type.is_zerobuf_type:
-            return '::zerobuf::fromJSON( ::zerobuf::getJSONField( json, "{0}" ), _{0} );'.\
-                format(self.cxxname)
+            return 'if( ::zerobuf::hasJSONField( json, "{0}" ))'\
+                   '{1}    ::zerobuf::fromJSON( ::zerobuf::getJSONField( json, "{0}" ), _{0} );'.\
+                format(self.cxxname, NEXTLINE)
         # convert enums to their name as a string
         if self.value_type.is_enum_type:
-            return 'set{0}( {1}( string_to_{1}( ::zerobuf::fromJSON< std::string >( ::zerobuf::getJSONField( json, "{2}" )))));'.\
-                format(self.cxxName, self.value_type.type, self.cxxname)
-        return 'set{0}( {1}( ::zerobuf::fromJSON< {2} >( ::zerobuf::getJSONField( json, "{3}" ))));'.\
-                format(self.cxxName, self.value_type.type, self.value_type.get_data_type(), self.cxxname)
+            return 'if( ::zerobuf::hasJSONField( json, "{2}" ))'\
+                   '{3}    set{0}( {1}( string_to_{1}( ::zerobuf::fromJSON< std::string >( ::zerobuf::getJSONField( json, "{2}" )))));'.\
+                format(self.cxxName, self.value_type.type, self.cxxname, NEXTLINE)
```

Comment:
It's not quiet obvious to me what the difference is between cxxName and cxxname. Any chance this can be clarified by finding a more appropriate variable name?